catboost/catboost_integration.py

```diff
@@ -0,0 +1,79 @@
+"""
+Optuna example that demonstrates a pruner for CatBoost.
+
+In this example, we optimize the validation accuracy of cancer detection using CatBoost.
+We optimize both the choice of booster model and their hyperparameters. Throughout
+training of models, a pruner observes intermediate results and stop unpromising trials.
+
+You can run this example as follows:
+    $ python catboost_integration.py
+
+"""
+
+import numpy as np
+import optuna
+
+import catboost as cb
+from sklearn.datasets import load_breast_cancer
+from sklearn.metrics import accuracy_score
+from sklearn.model_selection import train_test_split
+
+
+def objective(trial):
+    data, target = load_breast_cancer(return_X_y=True)
+    train_x, valid_x, train_y, valid_y = train_test_split(data, target, test_size=0.3)
```

Comment:
[minor comment] How about using the same `test_size` as [`lightgbm_integration.py`](https://github.com/optuna/optuna-examples/blob/6a6b20ad634627eebb3e7e104f73b70b45c6e624/lightgbm/lightgbm_integration.py) for consistency? Maybe we can address this part as another PR. (cc: @Crissman) 

```suggestion
    train_x, valid_x, train_y, valid_y = train_test_split(data, target, test_size=0.25)
```